glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -126,6 +137,8 @@ open class GleanInternalAPI internal constructor() {
     // Store the build information provided by the application.
     internal lateinit var buildInfo: BuildInfo
 
+    private var isCustomDataPath: Boolean? = null
```

Comment:
Any reason not to simplify things and just set this to `false` by default rather than making it an optional Boolean?

Reply:
I did this (at first) to guard any scenario where `isCustomDataPath` might be checked before it has a value set. I didn't want it to default to false in a scenario where it was still `null` and hadn't been checked yet.

But for the two places where I check it, it should be properly initialized by then, so I will make it non-optional.